api/kogito-events-core/src/main/java/org/kie/kogito/event/cloudevents/utils/BaseCloudEventValidator.java

```diff
@@ -0,0 +1,118 @@
+
+    protected abstract String getRfc2046Attribute();
+
+    final void validateCloudEvent(Map<String, Object> cloudEvent) throws InvalidCloudEventException {
+        List<String> errors = new ArrayList<>();
+
+        CloudEventUtils.getMissingAttributes(cloudEvent)
+                .forEach(missingAttribute -> errors.add("Missing mandatory attribute: " + missingAttribute));
+
+        validateNonEmptyAttributes(cloudEvent, errors);
+
+        validateRfc2046Attributes(cloudEvent, errors);
+
+        validateRfc3339Attributes(cloudEvent, errors);
+
+        if (!errors.isEmpty()) {
+            throw new InvalidCloudEventException(errors);
+        }
+    }
+
+    private static void validateAttribute(Map<String, Object> cloudEvent, String attribute, Predicate<Object> validation, String message, Collection<String> errors) {
+        Object value = cloudEvent.get(attribute);
+        if (value != null && validation.test(value)) {
+            errors.add(attribute + message);
+        }
+    }
+
+    /**
+     * Validates attributes that must adhere to the format specified in <a href="https://datatracker.ietf.org/doc/html/rfc3339">RFC 3339</a>.
+     *
+     * @param cloudEvent the CloudEvent
+     * @param errors the error list where the found errors should be put
+     */
+    private void validateRfc3339Attributes(Map<String, Object> cloudEvent, List<String> errors) {
+        validateAttribute(
+                cloudEvent,
+                getRfc3339Attribute(),
+                value -> !isRfc3339Value(value),
+                " MUST adhere to the format specified in RFC 3339 (https://datatracker.ietf.org/doc/html/rfc3339).",
+                errors);
+    }
+
+    /**
+     * Validates attributes that must adhere to the format specified in <a href="https://datatracker.ietf.org/doc/html/rfc2046">RFC 2046</a>.
+     *
+     * @param cloudEvent the CloudEvent
+     * @param errors the error list where the found errors should be put
+     */
+    private void validateRfc2046Attributes(Map<String, Object> cloudEvent, List<String> errors) {
+        validateAttribute(
+                cloudEvent,
+                getRfc2046Attribute(),
+                value -> !isRfc2046Value((String) value),
+                " MUST adhere to the format specified in RFC 2046 (https://datatracker.ietf.org/doc/html/rfc2046).",
+                errors);
+    }
+
+    private static boolean isRfc3339Value(Object value) {
+        if (value instanceof String) {
+            try {
+                DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse((String) value);
+                return true;
+            } catch (DateTimeParseException e) {
+                return false;
+            }
+        }
+
+        return false;
+    }
+
+    private static boolean isRfc2046Value(String value) {
+        if (value != null) {
+            Matcher matcher = RFCC2046.matcher(value);
+            return matcher.matches();
+        }
+        return false;
```

Comment:
The value is not going to be null
```suggestion
       return RFCC2046.matches(value).matches();
```